### ISSUE TITLE
Investigate HBase specific test failures

### ISSUE THREAD

**Issue by mbroecheler** [thinkaurelius/titan#1067]
So, the only thing for us to look into would be:

1)
GraphConstructionTest.shouldMaintainOriginalConfigurationObjectGivenToFactory:73->lambda$shouldMaintainOriginalConfigurationObjectGivenToFactory$100:74

which is probably related to some extra HBase options that are automagically inserted into the config but not copied and result in a discrepancy that this test checks for. I doubt this is a major issue.

2) 
HBaseGraphTest>TitanGraphTest.testIndexUpdatesWithReindexAndRemove:1460
HBaseGraphTest>TitanGraphTest.testGotGIndexRemoval:1239
which, based on the exception and the test themselves, likely have the same root cause.

Both of those last two HBaseGraphTest failures have the same exception
as their ultimate cause. I'm not sure what's going on here, and it's
worth a look:

```
Caused by: com.thinkaurelius.titan.diskstorage.TemporaryBackendException: Timed out waiting for next row data - storage error likely
        at com.thinkaurelius.titan.diskstorage.keycolumnvalue.scan.StandardScannerExecutor.run(StandardScannerExecutor.java:136)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by dalaro** [comment]
1. Still investigating.
2. The timeout was red herring.  The worker thread upon which the foreground thread had timed out had a totally different problem: buffer underrun in HBaseKCVS.RowIterator when attempting to convert a rowkey into a long.  This seems to be an old assumption in RowIterator that all rowkeys will be interpretable as longs, but this is not true when StandardScannerExecutor runs a removal job on graphindex.  This reinforces the need to propagate fatal background thread exceptions to foreground fulgora threads waiting on them (#983).  Removed this assumption (but did not fix #983) in f20489a9280c99a718b61e2569fe613ba16d2a0a.

**Comment by dalaro** [comment]
1. is fixed through f20489a9280c99a718b61e2569fe613ba16d2a0a (commit message has details)
